library/src/main/scala/scala/scalajs/runtime/EnvironmentInfo.scala

```diff
@@ -49,4 +49,20 @@ trait EnvironmentInfo extends js.Object {
   def sourceMapper: js.UndefOr[js.Function1[
     js.Array[JSStackTraceElem], js.Array[JSStackTraceElem]]] = js.native
 
+  /** Semantics configuration.
+   *
+   *  @group envInfo
+   */
+  val semantics: EnvironmentInfo.Semantics = js.native
```

Comment:
Yes, unfortunately. Now as I think of it, your change would not be binary compatible anyways: Linking with `<= 0.6.3` would cause `cast` to have `Fatal` semantics, irrespective of the linking semantics actually used.